### ISSUE TITLE
Cannot disable/remove a plugin. 

### ISSUE THREAD

**Issue by febbraro** [moby/moby#32609]
**Description**

I have installed the `rexray/ebs` plugin, and I was having difficultly getting it to work, so I was attempting to upgrade it from `rexray/ebs:latest` to `rexray/ebs:edge`. However, when I try to disable it I get the following error message:

`Error response from daemon: plugin rexray/ebs:latest is in use`

I do not have any volumes that were created with this plugin, so it seems that I should be able to disable/remove the plugin.

**Steps to reproduce the issue:**
1. docker plugin install rexray/ebs
2. docker plugin disable rexray/ebs

**Describe the results you received:**

It gave me an error message: `Error response from daemon: plugin rexray/ebs:latest is in use`

**Describe the results you expected:**

I expected it to disable the plugin so that I could then remove it.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 18:55:10 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Wed Apr  5 18:55:10 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 17.04.0-ce
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: pvt2k41nx4v3c98xa6ntm1h8v
 Is Manager: true
 ClusterID: udnrxhgowfyq8xycc685vjg1s
 Managers: 1
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.100.68
 Manager Addresses:
  192.168.100.68:2377
Runtimes: runc
Default Runtime: runc
Init Binary:
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-327.28.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.389GiB
Name: manager01
ID: O7BJ:7Z73:HAWX:TJKE:HTLC:FX7V:F5FN:6LVW:Y4NK:ET3P:46BX:CD7O
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior.
         Reformat the filesystem with ftype=1 to enable d_type support.
         Running without d_type support will not be supported in future releases.
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

I am running this on a CentOS 7 AWS EC2 t2.large instance.  I am also running this server as a Swarm mode manager in a 3 node Swarm mode cluster. 1 manager and 2 workers.

**Comment by febbraro** [comment]
@cpuguy83 Here is the issue we spoke about in Slack

**Comment by cpuguy83** [comment]
Thanks, it looks like if `Create()` fails we aren't dereferencing the plugin.
I'll work on this fix soon if someone else doesn't dibs it first.

**Comment by febbraro** [comment]
For those also having this problem, restarting the docker daemon allows you to disable the plugin.

**Comment by ripcurld0** [comment]
@cpuguy83 
was this ticket resolved?

**Comment by cpuguy83** [comment]
Hmm, I'm not sure.

**Comment by benoahriz** [comment]
I'm having this issue as well.   any workarounds available?

**Comment by cpuguy83** [comment]
I'm working on a fix. The fix itself is trivial, testing for it is complicated.

**Comment by cpuguy83** [comment]
Here we go: https://github.com/moby/moby/pull/35265

**Comment by FlorinAsavoaie** [comment]
Sorry for spamming here but is there any chance this could get pushed into the next release? This is really annoying.

**Comment by gjonespf** [comment]
Is there actually a workaround for this?  I see @febbraro mentioning restarting the docker daemon, however I don't believe that's possible with e.g. moby - restarting the VM simply restarts the plugin/remounts and you still can't disable or remove the plugin.  Alternatively, we wait for PR 308 I guess...
https://github.com/docker/docker-ce/pull/308

**Comment by larsskj** [comment]
I'm bitten by this one as well - now running Docker CE 17.12.0.

Still no workarounds?

Regards,
Lars

**Comment by cpuguy83** [comment]
@larsskj This is fixed in 17.12 (https://github.com/docker/docker-ce/commit/1c3f4cf12c1ec8a65e86da3d0da1a3308e0dcc7c#diff-38175530d7f7f615cad1b77a838a148c) , so most likely you have a different issue.
